### PR TITLE
Refactor the validation of Send/Amount step - Closes #669

### LINKED ISSUE #669: Refactor the validation of Send/Amount step
LSK has a static fee but BTC offers dynamic fees. this means that in Send page we should revalidate the input after the user changed the fee value. This requires a properly designed UI to reflect the effect of the fee on validation.

### PR DESCRIPTION
**Pull request by aydieneue**
# What was the bug or feature?
Described in #669 

### How did I fix it?
- Fixed translations with dynamic token type
- Updated `Send/Amount/LSK` component to use `DropdownHolder`
- Added `amount` and `balance` validator to `Send/Amount/BTC` component
- Updated `Send/Amount/BTC` component to use DropdownHolder

## Type of change
- Bug fix (a non-breaking change which fixes an issue)
- Enhancement (a non-breaking change which adds functionality)

### How to test it?
Try to trigger validation flow of LSK and BTC components by using invalid values for the amount.

# Checklist:
- My code follows the style guidelines of this project
- I have performed a self-review of my own code
- I have commented my code, particularly in hard-to-understand areas
- I have made corresponding changes to the documentation
- I have added tests that prove my fix is effective or that my feature works
- New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by reyraa** [comment]
The Send page throws because of a problem with followed accounts. It's not related to these changes, but I guess we have to fix it in order to be able to review this PR.

File: src/components/send/amount/btc.js [hunk 1]
```diff
@@ -71,76 +107,75 @@ class AmountBTC extends React.Component {
       amount: {
         value,
         normalizedValue,
-        validity: -1,
       },
     });
   }
 
   onChangeDynamicFee = (type) => {
```
Comment (reyraa) [inline src/components/send/amount/btc.js]: The correct name would be `onDynamicFeeChange`.

File: src/components/send/amount/btc.js [hunk 2]
```diff
@@ -71,76 +107,75 @@ class AmountBTC extends React.Component {
...
     const { priceTicker, settings: { currency } } = this.props;
-    const { amount: { value, normalizedValue } } = this.state;
+    const { amount } = this.state;
+    const amountValidity = this.validateAmount(amount);
 
     let valueInCurrency = 0;
 
-    if (value && this.validator(normalizedValue).code === 0 && priceTicker[currency]) {
-      valueInCurrency = (normalizedValue * priceTicker[currency]).toFixed(2);
+    if (amount.value && amountValidity.code === 0 && priceTicker[currency]) {
```
Comment (reyraa) [inline src/components/send/amount/btc.js]: Without `amount.value` we couldn't get `amountValidity.code === 0`. so I assume the later is enough.

File: src/components/send/amount/btc.js (hunk 1 above)
Comment (aydieneue) [reply src/components/send/amount/btc.js]: 👍 I agree with and also the method above should be renamed as `onAmountChange`.

I've taken a look at [React's official docs](https://reactjs.org/docs/forms.html#controlled-components) and they're actually using `handle` keyword for this kind of event listeners. 

We may consider to re-visit (or _define_ 😁) our naming convensions for cases like this.

**Proposal** 🧐
- `on + {verb}` or `handle + {verb}` for single input cases
- `on + {verb} + {property name}` or `handle + {verb} + {property name}` for multi-input cases

**Good** ✅
- `onChangeAmount` || `handleChangeAmount`
- `onChangeDynamicFee` || `handleChangeDynamicFee`
- `onSubmit` || `handleSubmit`
- `onForward`
- `onSend` || `onClickSend` || `handleClickSend`

**Bad** ❌
- [`setAddress`](https://github.com/LiskHQ/lisk-mobile/blob/development/src/components/send/recipient/index.js#L76-L87)
- [`submitForm`](https://github.com/LiskHQ/lisk-mobile/blob/development/src/components/send/recipient/index.js#L89-L110)
- [`forward`](https://github.com/LiskHQ/lisk-mobile/blob/development/src/components/send/recipient/index.js#L112-L126)
- [`send`](https://github.com/LiskHQ/lisk-mobile/blob/development/src/components/send/overview/index.js#L89-L114)
- [`openAcademy`](https://github.com/LiskHQ/lisk-mobile/blob/development/src/components/send/overview/index.js#L116-L120)

This standardization would also help us if we consider to use `Airbnb's JSX rules` in the project. Please [check this](https://github.com/airbnb/javascript/tree/master/react#ordering) for the reference. 

Thanks for the feedback 🙏🏻

**Pull request merged**